sound/soc/sof/sof-audio.h

```diff
@@ -387,6 +389,19 @@ struct snd_sof_widget {
 	int num_tuples;
 	struct snd_sof_tuple *tuples;
 
+	int num_sink_pins;
+	int num_source_pins;
```

Comment:
we should use uint32_t here, no reason to have negative pins.  Then you don't need to check num_source_pins < 0

Reply:
some analyzer tools always find some defect for int & uint usage.

Reply:
You are right, but we use int already for a lot of other values that can not be zero, for example, component id, pipeline_id, core_id, num_tuples, etc, we will need do repo level clean up. let's see opinion from others.

I think the only thing we need care is that a negative value of int will become a very big value in uint in reinterpret. in the real use of int and uint, most people will not think a lot in which one to use. and they just choose int.